core/domain/user_jobs_one_off.py

```diff
@@ -18,12 +18,50 @@
 
 from core import jobs
 from core.domain import subscription_services
+from core.domain import user_services
 from core.platform import models
 (exp_models, collection_models, feedback_models, user_models) = (
     models.Registry.import_models([
         models.NAMES.exploration, models.NAMES.collection,
         models.NAMES.feedback, models.NAMES.user]))
 
+class UserContributionsOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for creating and populating UserContributionsModels for 
+    all registered users that have contributed.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationSnapshotMetadataModel]
+
+    @staticmethod
+    def map(item):
+
+        split_id = item.id.split("-")
```

Comment:
Use rsplit() (see docs for _VERSION_DELIMITER in base_models.VersionedModel).

Better still -- Raine is working on adding a method to VersionedModel that does this splitting and returns the unversioned id; you could use that instead once her branch is merged in.